src/index.ts

```diff
@@ -69,14 +70,15 @@ export default function(settings?: ProblemMwSettings): Middleware {
         ctx.response.body.detail = detail;
       }
 
-      if (clientError) {
-        // tslint:disable-next-line no-console
-        console.warn(e);
-      } else {
-        // tslint:disable-next-line no-console
-        console.error(e);
+      if (settings) {
+        if (clientError && settings.quiet === false) {
+          // tslint:disable-next-line no-console
+          console.warn(e);
+        } else {
```

Comment:
```suggestion
      if (clientError && !quiet) {
          // tslint:disable-next-line no-console
          console.warn(e);
        } else {
```

Reply:
This patch probably breaks things, but this removes the extra nesting and the issue with  `settings` not being set.